### DISCUSSION TITLE
Create pages from json-data

### DISCUSSION THREAD

**Discussion by nils-a** [statiqdev/Discussions#176] [Q&A] [answered]
Hey people,
I have dreamed up some fancy idea, but I don't know how to get it working.

I have some json files in `input/json`, which are read in the `DATA` pipeline.
I can perfectly fetch them in `index.cshtml` using something like `@foreach (var doc in @Outputs.FromPipeline("Data")) ...` and render a nice overview table.

Now, additionally, I'd like to have one detail-page for every json file and also some pages that are grouped by a property (a bit like the archive page).

So, to visualize the output:

```
output/
├─ index [the overview page (this works fine)]
├─ details/
│  ├─ one file per json in `json/`
├─ locations/
│  ├─ one page per `location` property in the json containig data from all json files for the given location.

```

Do I have to write a custom pipeline/module for that or am I missing a hint in the docs?

**Comment by daveaglick** [comment, marked as answer]
This is pretty much exactly what Discover .NET and the .NET Foundation sites do for some of their data-heavy portions of the site. Both of those use custom pipelines to accomplish it, but I think you can probably even do it with the [archives](https://www.statiq.dev/guide/web/archives) feature in Statiq Web. Let me see if I can work up an example for you - it just might take me a few days to get to it.

FWIW, this sort of thing is pretty much _exactly_ why Statiq exists in the first place. I found doing this kind of data manipulation in other generators was too confusing and/or impossible. So I'm hopeful there's a quick and easy way to accomplish it, or what am I even doing 😂

**Reply by nils-a** [reply]
`ArchivePipelines` was the fact I was missing. I was trying to access the `Data` pipeline using solely `ArchiveSources`.

So, now It's a real simple case of:

```razor
ArchivePipelines: Data
ArchiveKey: => GetString("location")
Title: Locations
DestinationPath: "locations/index.html"
---
<div>
@if (Document.ContainsKey(Keys.GroupKey))
{
  <div>
    @* item template *@
  </div>
}
else
{
  <div>
    @* overview template *@
  </div>
}  
</div>
```

**Reply by daveaglick** [reply]
Yes, exactly what I was thinking! The idea with archives is that you can slice JSON or other data-heavy documents in all kinds of different ways right from templates without having to create new pipelines or modules. Another feature I really need to do a better job of explaining and promoting.
